File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/analytics/model/ContentAnalyticsActionEvent.java
package org.cru.godtools.tract.analytics.model;

import com.google.common.base.Strings;

import org.cru.godtools.analytics.model.AnalyticsActionEvent;
import org.cru.godtools.analytics.model.AnalyticsSystem;
import org.cru.godtools.xml.model.AnalyticsEvent;

import java.util.Map;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class ContentAnalyticsActionEvent extends AnalyticsActionEvent {
    @NonNull
    private final AnalyticsEvent mAnalyticsEvent;

    public ContentAnalyticsActionEvent(@NonNull final AnalyticsEvent event) {
        mAnalyticsEvent = event;
    }

    @Override
    public boolean isForSystem(@NonNull final AnalyticsSystem system) {
        return mAnalyticsEvent.isForSystem(system);
    }

    @NonNull
    @Override
    public String getAction() {
        return Strings.nullToEmpty(mAnalyticsEvent.getAction());
    }

    @Nullable
    @Override
    public Map<String, ?> getAttributes() {
        return mAnalyticsEvent.getAttributes();
    }
}

File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/analytics/model/ContentAnalyticsActionEvent.kt
package org.cru.godtools.tract.analytics.model

import androidx.annotation.VisibleForTesting
import org.cru.godtools.analytics.model.AnalyticsActionEvent
import org.cru.godtools.analytics.model.AnalyticsSystem
import org.cru.godtools.xml.model.AnalyticsEvent

class ContentAnalyticsActionEvent(@VisibleForTesting internal val event: AnalyticsEvent) :
    AnalyticsActionEvent(action = event.action.orEmpty()) {
    override fun isForSystem(system: AnalyticsSystem) = event.isForSystem(system)

    override val firebaseParams get() = event.attributes
}
